storage/watchable_store_bench_test.go

```diff
@@ -81,3 +81,39 @@ func BenchmarkWatchableStoreUnsyncedCancel(b *testing.B) {
 		cancels[idx]()
 	}
 }
+
+func BenchmarkWatchableStoreSyncedCancel(b *testing.B) {
+	s := newWatchableStore(tmpPath)
+	defer func() {
+		s.store.Close()
+		os.Remove(tmpPath)
+	}()
+
+	// Put a key so that we can spawn watchers on that key
+	testKey := []byte("foo")
+	testValue := []byte("bar")
+	s.Put(testKey, testValue)
+
+	w := s.NewWatcher()
+
+	// put 1 million watchers on the same key
+	const watcherSize = 200000
```

Comment:
I set 200k because it was hanging too long for slice implementation. Will change it for new implementation.